app/src/main/cpp/skyline/services/hid/IHidServer.cpp

```diff
@@ -105,4 +108,34 @@ namespace skyline::service::hid {
         state.input->npad.at(id).SetAssignment(NpadJoyAssignment::Dual);
         state.input->npad.Update();
     }
+
+    void IHidServer::CreateActiveVibrationDeviceList(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        manager.RegisterService(SRVREG(IActiveVibrationDeviceList), session, response);
+    }
+
+    void IHidServer::SendVibrationValues(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        request.Skip<u64>(); // appletResourceUserId
+
+        auto &handleBuf = request.inputBuf.at(0);
+        std::span handles(reinterpret_cast<NpadDeviceHandle *>(handleBuf.address), handleBuf.size / sizeof(NpadDeviceHandle));
+        auto &valueBuf = request.inputBuf.at(1);
+        std::span values(reinterpret_cast<NpadVibrationValue *>(valueBuf.address), valueBuf.size / sizeof(NpadVibrationValue));
+
+        for (int i = 0; i < handles.size(); ++i) {
+            auto &handle = handles[i];
+
+            auto &device = state.input->npad.at(handle.id);
+            if (device.type == handle.GetType()) {
+                if (i + 1 != handles.size() && handles[i + 1].id == handle.id && handles[i + 1].isRight && !handle.isRight) {
+                    state.logger->Info("Vibration #{}&{} - Handle: 0x{:02X} (0b{:05b}), Vibration: {:.2f}@{:.2f}Hz, {:.2f}@{:.2f}Hz - {:.2f}@{:.2f}Hz, {:.2f}@{:.2f}Hz", i, i + 1, u8(handle.id), u8(handle.type), values[i].amplitudeLow, values[i].frequencyLow, values[i].amplitudeHigh, values[i].frequencyHigh, values[i + 1].amplitudeLow, values[i + 1].frequencyLow, values[i + 1].amplitudeHigh, values[i + 1].frequencyHigh);
```

Comment:
Having this as info 100% isn't a good idea, drop it down to debug.

Or alternatively add a new log level so, trace, so there isn't so much SVC garbage in debug

Reply:
This was meant to be a temporary thing while I debugged, didn't mean to leave it in. I think this is something that'll be solved by our newer logger regardless so it shouldn't be an issue.